Detectors/MUON/MCH/Simulation/include/MCHSimulation/Digitizer.h

```diff
@@ -65,14 +73,21 @@ class Digitizer
   // digit per pad
   std::vector<Digit> mDigits;
 
+  //map between index and detector-element-ID
   std::map<int, int> mdetID;
-
+  //segmentation for each detector-element
   std::vector<mapping::Segmentation> mSeg;
 
+  //MCLabel container (transient)
+  o2::dataformats::MCTruthContainer<o2::MCCompLabel> mMCTruthContainer;
+  //MCLabel container (output): why needed?
+  o2::dataformats::MCTruthContainer<o2::MCCompLabel> mMCTruthOutputContainer;
   //member with parameters and signal generation
-  Response mMuonresponse;
+  Response mMuonresponse_stat1{ 0 };//station 1
```

Comment:
I'm still not at ease with the current "entanglement" of Digitizer and Response classes. I would either remove (i.e. hide) the Response (which is currently a concrete class) from the Digitizer _interface_ or make the Response abstract (but with only one implementation so far = Mathieson, I would not go that route for now).
 
 One way to hide it is simply to have a free function in Digitizer.cxx returning to whoever needs it the response information. BTW, I would also consider extracting from the response the bare bone functionality, i.e. the "spreading" of a unit charge originating at position (x,y) onto a box (x1,y1,x2,y2) (the current response.chargePad, but w/o the charge, which is only a multiplicative factor in the end anyway).

Reply:
For the 2nd part, I factorise this out now of the calculation.

Reply:
Hi @aphecetche 

Sorry for my ignorance, but I am not sure, whether I understand the free function advice properly: 
- I am using in Digitizer.cxx 6 different functions of the members of type Response.

Do you want that I write for each of these functions another free function taking as input a response object and, the parameter that is needed and which basically just does the same as the member of Response itself, i.e. free functions of the type like:
namespace o2{
namespace mch{
class Digitizer{
...
};
double etocharge(Response response, double charge) { response.etocharge(charge) };
}
}

What do I gain in terms of encapsulation, if i do something like that apart from the fact that the things can be written all in the same spot as in-line code? I think that I am missing what you want me to do :) . 
Alternatively, I can 'hide' part of the processing in the Response itself to reduce the number of different functions that have to be called, but I would find it better that in the digitiser itself one sees the different steps and the coordinate transform and not just  a blackbox that gives back the charge. Of course, one could write yet another layer between the Response and the digtizer for this "workflow" part, but the code is not very long anyhow. Or I write another class for just the charge conversion and a few other parameters, but I would find that also a bit overkill to have only the Mathieson integration into the Response class. But of course can be done.

Reply:
@aphecetche 
Thanks a lot. I read through, removed one remaining isStation bool and took it in.